tests/Expr/CompositeExpressionTest.php

```diff
@@ -15,12 +15,26 @@
 class CompositeExpressionTest extends TestCase
 {
     /** @return list<array{type:string, expressions: list<mixed>}> */
-    public function invalidDataProvider(): array
+    public static function invalidDataProvider(): array
     {
         return [
             ['type' => CompositeExpression::TYPE_AND, 'expressions' => [new Value('value')]],
             ['type' => CompositeExpression::TYPE_AND, 'expressions' => ['wrong-type']],
-            ['type' => CompositeExpression::TYPE_NOT, 'expressions' => [$this->createMock(Expression::class), $this->createMock(Expression::class)]],
+            [
+                'type' => CompositeExpression::TYPE_NOT,
+                'expressions' => [
+                    new class () implements Expression {
+                        public function visit(ExpressionVisitor $visitor)
+                        {
+                        }
+                    },
+                    new class () implements Expression {
+                        public function visit(ExpressionVisitor $visitor)
+                        {
+                        }
+                    },
+                ],
+            ],
```

Comment:
You could use `self::createStub()` to replace `$this->createMock()` in a static method.

Reply:
TIL `createMock` is not statick, but `createStub` is :bulb